lldb/test/API/lang/swift/async/tasks/TestSwiftTaskSelect.py

```diff
@@ -0,0 +1,75 @@
+import lldb
+from lldbsuite.test.decorators import *
+from lldbsuite.test.lldbtest import TestBase
+import lldbsuite.test.lldbutil as lldbutil
+
+
+class TestCase(TestBase):
+
+    def test_backtrace_selected_task(self):
+        self.build()
+        lldbutil.run_to_source_breakpoint(
+            self, "break here", lldb.SBFileSpec("main.swift")
+        )
+        self.runCmd("language swift task select task")
+        self.expect(
+            "thread backtrace",
+            substrs=[
+                ".sleep(",
+                "`second() at main.swift:6:",
+                "`first() at main.swift:2:",
+                "`closure #1 in static Main.main() at main.swift:12:",
+            ],
+        )
+
+    def test_navigate_selected_task_stack(self):
+        self.build()
+        # target, process, thread, bkpt
+        _, process, _, _ = lldbutil.run_to_source_breakpoint(
+            self, "break here", lldb.SBFileSpec("main.swift")
+        )
+        self.runCmd("language swift task select task")
+
+        self.expect(
+            "thread list",
+            substrs=[
+                "* thread #4294967295: tid = ",
+                "libswift_Concurrency.",
+                "._sleep(",
+            ],
+        )
+
+        frame_index = -1
+        for frame in process.selected_thread:
+            if "`second()" in str(frame):
+                frame_index = frame.idx
+        self.assertNotEqual(frame_index, -1)
+
+        self.expect(
```

Comment:
I would feel better about this test if it would use the SBAPI instead of matching for the precise formatting of the frame rendering. Otherwise we'll need to update this every tome some changes the UI upstream.

Reply:
something like getting the current frame from the current thread and checking SBFrame::GetFunctionName() etc ...

Reply:
done